frontend/src/views/admin/matching/import-export.tsx

```diff
@@ -0,0 +1,442 @@
+        Record<string, string | null>
+    >({});
+
+    const matchingData = useSelector(matchingDataSelector);
+
+    function _onFileChange(event: React.ChangeEvent<HTMLInputElement>) {
+        if (!event.target || !event.target.files) {
+            return;
+        }
+        const file = event.target.files[0];
+        setFileInputLabel(file.name);
+
+        const reader = new FileReader();
+        reader.onload = (e) => {
+            if (!e.target || typeof e.target.result === "string") {
+                console.warn(
+                    "File of unexpected type",
+                    typeof e.target?.result
+                );
+                return;
+            }
+            setFileArrayBuffer(e.target.result);
+        };
+        reader.readAsArrayBuffer(file);
+    }
+
+    React.useEffect(() => {
+        if (!fileArrayBuffer) {
+            return;
+        }
+
+        const rawData = new Uint8Array(fileArrayBuffer);
+        try {
+            const str = new TextDecoder().decode(rawData);
+            setFileContent({ data: JSON.parse(str), fileType: "json" });
+            return;
+            // eslint-disable-next-line
+        } catch (e) {}
+
+        console.warn(
+            "Could not determine file type for",
+            fileInputLabel,
+            fileArrayBuffer
+        );
+    }, [fileArrayBuffer, fileInputLabel]);
+
+    React.useEffect(() => {
+        if (!fileContent) {
+            return;
+        }
+        try {
+            if (Object.keys(fileContent.data).includes("matches")) {
+                // Filter matches based on whether
+                // - the utorid and positionId combo don't exist at all in the current matching data set, or
+                // - the combination exists but the assignment status or hoursAssigned is different
+                const diffedMatches = fileContent.data.matches.filter(
+                    (newMatch: Match) =>
+                        !matchingData.matches.find(
+                            (oldMatch: Match) =>
+                                oldMatch.utorid === newMatch.utorid &&
+                                oldMatch.positionCode === newMatch.positionCode
+                        ) ||
+                        matchingData.matches.find(
+                            (oldMatch: Match) =>
+                                oldMatch.utorid === newMatch.utorid &&
+                                oldMatch.positionCode ===
+                                    newMatch.positionCode &&
+                                oldMatch.status !== "assigned" &&
+                                (oldMatch.status !== newMatch.status ||
+                                    oldMatch.hoursAssigned !==
+                                        newMatch.hoursAssigned)
+                        )
+                );
+
+                setDiffedMatches(diffedMatches);
+            }
+
+            if (Object.keys(fileContent.data).includes("guarantees")) {
```

Comment:
Just do `if (fileContent.data["guarantees"]) ...` The same comment applies everywhere you do this.